kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs

```diff
@@ -1114,7 +1303,11 @@ impl<'tcx> GotocCtx<'tcx> {
                         }
                     }
                 }
-                Variants::Multiple { tag_encoding, variants, .. } => {
+                Variants::Multiple { tag_encoding, variants, tag_field, .. } => {
+                    // Contrary to generators, currently enums have only one field (the discriminant), the rest are in the variants:
+                    assert!(layout.fields.count() <= 1);
+                    // Contrary to generators, the discriminant is the first (and only) field for enums:
+                    assert_eq!(*tag_field, 0);
```

Comment:
What happens in a Niche encoding

Reply:
It seems that they have a `tag_field` anyway: https://doc.rust-lang.org/stable/nightly-rustc/rustc_target/abi/enum.Variants.html#variant.Multiple.field.tag_field. So this assertion should not be a problem (and the regression tests pass).